Workflow/Sources/AnyWorkflow.swift

```diff
@@ -131,14 +131,18 @@ extension AnyWorkflow {
             return AnyWorkflow<Rendering, NewOutput>.Storage<T>(
                 workflow: workflow,
                 renderingTransform: renderingTransform,
-                outputTransform: { transform(self.outputTransform($0)) }
+                outputTransform: { [outputTransform] in
+                    transform(outputTransform($0))
+                }
```

Comment:
in the common use case, you have things like:

```swift
MyWorkflow()
  .asAnyWorkflow()  // creates internal storage
  .mapOutput { ... }  // creates new internal storage & would previously hold a reference to old storage
  .rendered(...)
```

now we won't keep the storage instances around longer than necessary.